lib/src/rules/directives_ordering.dart

```diff
@@ -110,22 +143,46 @@ class DirectivesOrdering extends LintRule {
     if (node == null) {
       return;
     }
+    if (_visitor.lintedNodes.contains(node)) {
+      return;
+    }
+    _visitor.lintedNodes.add(node);
```

Comment:
`Set.add` returns `true` if the set was modified (if the argument was not previously an element, so it's more efficient to write
```
if (_visitor.lintedNodes.add(node)) {
  reporter.report...
}
```